### ISSUE TITLE
Autocomplete: allowOtherValues does not work

### ISSUE THREAD

**Issue by harish-prakash** [segmentio/evergreen#1562]
> <video src="https://user-images.githubusercontent.com/32839065/205111940-ac5b057a-25a1-4d86-b62e-bc49ab2ae6b4.mov"/>

**Steps to reproduce**
Use the `<Autocomplete/>` component with `allowOtherValues` prop.
```tsx
const Page = () => {
  const [value, setValue] = useState('')
  return (
    <div>
      <Autocomplete
        allowOtherValues
        id="autocomplete-test"
        onChange={setValue}
        items={[
          'red',
          'green',
          'yellow',
          'this does not render on the textbox but sets the state correctly'
        ]}
      >
        {({ getInputProps, getRef, openMenu, inputValue }) => {
          return (
            <TextInput
              ref={getRef}
              {...getInputProps({ onFocus: openMenu })}
              value={inputValue}
            />
          )
        }}
      </Autocomplete>
      <Button onClick={() => console.debug('value... ', value)}>
        Log Value
      </Button>
    </div>
  )
}
```

**Comment by brandongregoryscott** [comment]
Hi @harish-prakash, what version are you using? I believe this was fixed with #1496 / https://github.com/segmentio/evergreen/releases/tag/v6.10.4. Here's a CodeSandbox with your example code: https://codesandbox.io/s/issue-1562-38zjfj?file=/src/App.tsx

**Comment by harish-prakash** [comment]
I see that this was fixed in 6.10.4, I am using 6.10.3 🤦‍♂️. Closing this issue 🙌
